### PR TITLE
First Person Animations

### PR DESCRIPTION
**Pull request by ZsoltMolnarrr**
Usage:

```
// Given `ModifierLayer layer`

// Create a normal player
var player = new KeyframeAnimationPlayer(copy.build(), 0);

// Instruct it to play animation in first person with given configuration
player.playInFirstPersonAsCombat(new FirstPersonAnimation.Configuration(
  true, // Show right arm
  true, // Show left arm
  true, // Show right item
  true // Show left item
));

// Start the player normally
layer.replaceAnimationWithFade(AbstractFadeModifier.standardFadeIn(fadeIn, Ease.INOUTSINE), player);
```

Checking if first person model is being rendered, during a render cycle:
```
if (FirstPersonRenderState.isRenderCycleFirstPerson()) {
  // ...
}
```

Additionally we can override this function in `KeyframeAnimationPlayer` to allow partial playback duration
```
@Nullable
public FirstPersonAnimation getActiveFirstPersonAnimation(float tickDelta)
```

### CODE REVIEW

File: coreLib/src/main/java/dev/kosmx/playerAnim/api/layered/AnimationStack.java [hunk 1]
```diff
@@ -88,4 +91,24 @@ public boolean removeLayer(int layerLevel) {
         return layers.removeIf(integerIAnimationPair -> integerIAnimationPair.getLeft() == layerLevel);
     }
 
+    @Nullable
+    public FirstPersonAnimation getActiveFirstPersonAnimation(float tickDelta) {
+        for (Pair<Integer, IAnimation> layer : layers) {
+            IAnimation animation = layer.getRight();
+            if (animation instanceof ModifierLayer) {
+                animation = ((ModifierLayer)animation).getAnimation();
+            }
```
Comment (ZsoltMolnarrr) [inline coreLib/src/main/java/dev/kosmx/playerAnim/api/layered/AnimationStack.java:100]: This works for my cases, but doesn't look like a good universal solution.

File: minecraft/common/src/main/resources/playerAnimator-common.mixins.json [hunk 2]
```diff
@@ -9,11 +9,16 @@
     "BipedEntityModelMixin",
     "FeatureRendererMixin",
     "HeldItemMixin",
+    "LivingEntityRenderRedirect_bendOnly",
```
Comment (ZsoltMolnarrr) [inline minecraft/common/src/main/resources/playerAnimator-common.mixins.json:12]: Automatically sorted by IDE

File: coreLib/src/main/java/dev/kosmx/playerAnim/api/first_person/FirstPersonRenderState.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package dev.kosmx.playerAnim.api.first_person;
+
+
+public class FirstPersonRenderState {
```
Comment (ZsoltMolnarrr) [inline coreLib/src/main/java/dev/kosmx/playerAnim/api/first_person/FirstPersonRenderState.java:4]: This is important to stay in the API package.
Checking for ongoing render cycle state can be important for some features.

File: coreLib/src/main/java/dev/kosmx/playerAnim/api/first_person/FirstPersonAnimation.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
+package dev.kosmx.playerAnim.api.first_person;
+
+import dev.kosmx.playerAnim.api.layered.IAnimation;
+
+public final class FirstPersonAnimation {
```
Comment (ZsoltMolnarrr) [inline coreLib/src/main/java/dev/kosmx/playerAnim/api/first_person/FirstPersonAnimation.java:5]: This used to be a record, now its a class, so it looks much bigger.

File: coreLib/src/main/java/dev/kosmx/playerAnim/api/layered/AnimationStack.java (hunk 1 above)
Comment (KosmX) [reply coreLib/src/main/java/dev/kosmx/playerAnim/api/layered/AnimationStack.java:100]: I'll definitely refactor this.

File: minecraft/common/src/main/resources/playerAnimator-common.mixins.json (hunk 2 above)
Comment (KosmX) [reply minecraft/common/src/main/resources/playerAnimator-common.mixins.json:12]: okay...

**Comment by KosmX** [comment]
Thank you, it has been copied without merging.

**Pull request closed without merging**